CRM/Contact/BAO/Contact/Location.php

```diff
@@ -40,16 +40,22 @@ class CRM_Contact_BAO_Contact_Location {
    *
    * @param bool $isPrimary
    * @param int $locationTypeID
+   * @param bool $isBilling
    *
    * @return array
    *   Array of display_name, email, location type and location id if found, or (null,null,null, null)
    */
-  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL) {
+  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL, $isBilling = FALSE) {
     $primaryClause = NULL;
     if ($isPrimary) {
       $primaryClause = " AND civicrm_email.is_primary = 1";
     }
 
+    // CRM-20362 retrieving Billing email address.
+    if ($isBilling) {
+      $primaryClause = " AND civicrm_email.is_billing = 1";
```

Comment:
hmm yeh i guess so, I just figure that if someone has say a work address as primary and have another address as billing then they may expect communication to their work address. 

I'm not overall concerned about exposing is_billing as I think Agileware made the point in the comment below. 

One thing would be is what if you have 2 billing email addresses? as that is something i know the GUI would allow